src/lib/datatypes/sol-buffer.c

```diff
@@ -38,18 +38,25 @@
 #include "sol-log.h"
 #include "sol-util.h"
 
+SOL_ATTR_PURE static inline size_t
+nul_byte_size(struct sol_buffer *buf)
+{
+    return (buf->flags & SOL_BUFFER_FLAGS_NO_NUL_BYTE) ? 0 : 1;
+}
+
 SOL_API int
 sol_buffer_resize(struct sol_buffer *buf, size_t new_size)
 {
     char *new_data;
 
     SOL_NULL_CHECK(buf, -EINVAL);
-    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_FIXED_CAPACITY, -EPERM);
-    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_NO_FREE, -EPERM);
 
     if (buf->capacity == new_size)
         return 0;
 
+    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_FIXED_CAPACITY, -EPERM);
+    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_NO_FREE, -EPERM);
+
     new_data = realloc(buf->data, new_size);
```

Comment:
These checks were moved, not removed. Shouldn't make a difference if old_size == new_size. (Maybe the move isn't necessary anymore, will check before pushing.)